### PR TITLE
Add run DELETE method

### PR DESCRIPTION
**Pull request by ushergod**
Signed-off-by: Sihan Wang <wsihan@vmware.com>

### CODE REVIEW

File: column/api/controller/run_controller.py [hunk 1]
```diff
@@ -55,14 +57,23 @@ class Run(flask_restful.Resource):
...
-        return run
+        return run_model.format_response(run)
 
+    def delete(self, id):
+        """Delete run by id"""
+        run = self.backend_store.get_run(id)
+        if not run:
+            return abort(404, message="Run {} doesn't exist".format(id))
+        if not self.manager.delete_run(run):
+            return abort(400, message="Fail to find the task queue"
```
Comment (ericwb) [inline column/api/controller/run_controller.py]: s/Fail/Failed

File: column/api/manager/run_manager.py [hunk 2]
```diff
@@ -92,8 +94,23 @@ def _run_playbook(self, run):
...
         t = threading.Thread(target=self._run_playbook, args=[run])
         t.start()
-        return run_info
+        return run
+
+    def delete_run(self, run):
+        timeout = 10
+        counter = 0
+        while counter < timeout:
+            if run['api_runner'].pbex:
```
Comment (ushergod) [inline column/api/manager/run_manager.py]: need checking when task is completed

File: column/api/manager/run_manager.py [hunk 3]
```diff
@@ -3,13 +3,15 @@
 
 import copy
 import logging
+import time
```
Comment (ericwb) [inline column/api/manager/run_manager.py]: alpha sort. time comes after threading

File: column/api_runner.py [hunk 4]
```diff
@@ -43,6 +43,7 @@ class APIRunner(runner.Runner):
     def __init__(self, inventory_file=None, **kwargs):
         super(self.__class__, self).__init__(inventory_file, **kwargs)
         self._callbacks = []
+        self.pbex = None
```
Comment (ericwb) [inline column/api_runner.py]: Why keep the PlaybookExecutor and not just the task queue manager?

File: column/api/manager/run_manager.py [hunk 5]
```diff
@@ -92,8 +94,23 @@ def _run_playbook(self, run):
...
+                run_model.format_response(run))
         t = threading.Thread(target=self._run_playbook, args=[run])
         t.start()
-        return run_info
+        return run
+
+    def delete_run(self, run):
+        timeout = 10
+        counter = 0
+        while counter < timeout:
```
Comment (ericwb) [inline column/api/manager/run_manager.py]: Much better to use a for loop with index

for i in xrange(10):

File: column/api/model/run_model.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+from collections import defaultdict
+
+
+def get_progress(run):
+    # TODO(ushergod) need to create backend model class and
+    # put this fucntion into the model class
```
Comment (ericwb) [inline column/api/model/run_model.py]: s/fucntion/function

File: column/api/model/run_model.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    progress = run['api_runner'].get_progress()
+    return 0 if progress is None else progress
+
+
+def format_response(run):
+    # TODO(ushergod) need to create backend model class and
+    # put this fucntion into the model class
```
Comment (ericwb) [inline column/api/model/run_model.py]: s/fucntion/function

**Pull request merged**